From e933e819e8556bb1c2709deb1e71535a37dc152c Mon Sep 17 00:00:00 2001
From: Anuj Aggarwal <anuj.aggarwal@ti.com>
Date: Tue, 9 Jun 2009 12:35:32 +0530
Subject: [PATCH 424/426] Audio: Adding PM code in audio driver

Modification is required to make the audio driver successfully work
after the system wakes from the off mode.
---
 sound/soc/omap/omap-pcm.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/sound/soc/omap/omap-pcm.c b/sound/soc/omap/omap-pcm.c
index 6454e15..202ff53 100644
--- a/sound/soc/omap/omap-pcm.c
+++ b/sound/soc/omap/omap-pcm.c
@@ -29,6 +29,7 @@
 #include <sound/soc.h>
 
 #include <mach/dma.h>
+#include <mach/pm.h>
 #include "omap-pcm.h"
 
 static const struct snd_pcm_hardware omap_pcm_hardware = {
@@ -192,6 +193,7 @@ static int omap_pcm_trigger(struct snd_pcm_substream *substream, int cmd)
 	case SNDRV_PCM_TRIGGER_RESUME:
 	case SNDRV_PCM_TRIGGER_PAUSE_RELEASE:
 		prtd->period_index = 0;
+		omap2_block_sleep();
 		omap_start_dma(prtd->dma_ch);
 		break;
 
@@ -200,6 +202,7 @@ static int omap_pcm_trigger(struct snd_pcm_substream *substream, int cmd)
 	case SNDRV_PCM_TRIGGER_PAUSE_PUSH:
 		prtd->period_index = -1;
 		omap_stop_dma(prtd->dma_ch);
+		omap2_allow_sleep();
 		break;
 	default:
 		ret = -EINVAL;
-- 
1.6.2.4

